Do not crash the node on unimplemented methods

Geod24 opened the pull request:
There are legitimate use case for nodes that get called with the wrong method.
In the real world, this results in a 404 error being returned.
In Localrest it used to crash the remote node because of the assert, but no more.

Geod24 committed:
Do not crash the node on unimplemented methods

There are legitimate use case for nodes that get called with the wrong method.
In the real world, this results in a 404 error being returned.
In Localrest it used to crash the remote node because of the assert, but no more.

Geod24 commented:
Fixes #79

AndrejMitrovic commented on source/geod24/LocalRest.d:
```diff
@@ -475,7 +475,9 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
-            assert(0, "Unmatched method name: " ~ cmd.method);
+            C.trySend(cmd.sender,
+                      Response(Status.Failed, cmd.id,
+                               SerializedData("Method not implemented (404)")));
```
Would it be of any use to emit the method name in the message?

AndrejMitrovic replied:
Hmm actually it would make even more sense to throw a custom 404 exception. If vibe.d does something similar we could handle both in our agora codebase.

Geod24 committed:
Trivial: Clean up some imports

Geod24 committed:
Make client (request) error a separate type of error

Some client logic might rely on treating errors differently,
e.g. HTTP has 4xx and 5xx error codes,
where 4xx could be used to drive the application logic.
While we don't expose errors the same way as HTTP,
we have a use case for having a separate kind of `Exception` for client errors,
the first of which would be a syntactically invalid query.

Geod24 commented:
Altered the code to throw a "client error" when there is one.

linked0 reviewed (approved):
LGTM.
I checked the modified code and approved it. I need to skim over the whole code though.
